swift-sdk/Internal/AuthManager.swift

```diff
@@ -35,7 +37,13 @@ class AuthManager: IterableInternalAuthManagerProtocol {
     }
     
     // @objc attribute only needed for the pre-iOS 10 Timer constructor in queueAuthTokenExpirationRefresh
-    @objc func requestNewAuthToken() {
+    @objc func requestNewAuthToken(_ hasFailedPriorAuth: Bool = false) {
+        guard !self.hasFailedPriorAuth || !hasFailedPriorAuth else {
```

Comment:
Yeah, it's designed so that`AuthManager` keeps track of when **not** to retry auth retrieval, like if `self.hasFailedPriorAuth` is already `true`, then, a `requestNewAuthToken` call with `hasFailedPriorAuth` as `true` will prevent it from executing the token retrieval code.

On top of that, if it gets called with `hasFailedPriorAuth` as `false`, it needs to reset that flag and run the retrieval code, and **not** prevent subsequent calls.

Reply:
Seems like new code will be not be fetched only when both values are true. It will be fetched in all other cases.
That seems fine. 
Can you please add a test that verifies this? I don't see any test calling `requestNewAuthToken(hasFailedPriorAuth: true)`. We should probably verify that this will result in a fetch only if self.hasFailedPriorAuth has failed.